### DISCUSSION TITLE
torch_geometric.loader batch a hetero graph into subgraphs based on user provided list of nodes and edges

### DISCUSSION THREAD

**Discussion by mank-hub** [pyg-team/pytorch_geometric#7277] [Q&A]
How to batch a hetero graph or a homogeneous graph into subgraphs for training based on user provided list of nodes and edges for each subgraph. There are various methods which do an automatic sampling, but they don't fit my case.

**Comment by rusty1s** [comment]
Wouldn't `NeighborLoader` work in this case if you specify the nodes you are interested in via `input_nodes` argument?
